Reordering hidden columns

johanrd opened the pull request:
See https://github.com/CrowdStrike/ember-headless-table/pull/185 for reference

NullVoxPopuli commented:
how'd you pull over the git history? that's cool :tada:

NullVoxPopuli commented:
Labeling as breaking, since the API for ColumnOrder is changing

NullVoxPopuli commented:
oh is still WIP? I haven't reviewed cause i thought it was still being worked on

johanrd commented:
Ah, sure, np. I just tested this properly now, and think this is ready for review :pray:.

Be ware: What is not working well (sometimes) is saving preferences to localstore/etc when using localhost. That is not related to this PR though, see https://github.com/CrowdStrike/ember-headless-table/pull/239 and https://github.com/CrowdStrike/ember-headless-table/pull/237

NullVoxPopuli commented on docs-app/public/docs/3-demos/external-column-ordering.md:
```diff
@@ -27,7 +27,11 @@ export default class extends Component {
     this.pendingColumnOrder = new ColumnOrder({
-      columns: () => this.columns,
+      allColumns: () => this.columns,
+      availableColumns: () => this.columns.reduce(function(acc, col) {
```
sorry it's taken me so long to get to this PR. <3


can you add a comment in this demo's code (maybe inline) about why `ColumnOrder` takes two separate collections of the columns?

johanrd commented:
@NullVoxPopuli Thanks, I have now addressed your comments. I have managed to find a way to make `availableColumns` optional in the ColumnOrder args (while making the removal/add columns from storage more resilient), so this may not be a breaking change anymore.?
